src/validation.cpp

```diff
@@ -1206,10 +1321,21 @@ PackageMempoolAcceptResult MemPoolAccept::AcceptMultipleTransactions(const std::
 
     std::vector<Workspace> workspaces{};
     workspaces.reserve(txns.size());
-    std::transform(txns.cbegin(), txns.cend(), std::back_inserter(workspaces),
-                   [](const auto& tx) { return Workspace(tx); });
+    std::transform(txns.cbegin(), txns.cend(), std::back_inserter(workspaces), [this](const auto& tx) {
+        return Workspace(tx, m_limit_descendants, m_limit_descendant_size);
+    });
     std::map<const uint256, const MempoolAcceptResult> results;
 
+    if (const auto v3_violation{CheckV3Inheritance(txns)}) {
```

Comment:
I think this is broken, it's only considering the deduplicated package, which "blinds" this evaluation from checking if the in-mempool parent is V3. Hence, the child as V2 would be accepted, even if parent in mempool was V3.

Reply:
Also see `CheckV3Inheritance` call on L932

Reply:
Ok is this a quick check that misses these scenarios, then `PreCheck` later calls it again with all ancestors?

Reply:
ah ok, the mempool lock comment. got it!